### DISCUSSION TITLE
Automatic suppressing keeping car awake → Automatic Travel Mode

### DISCUSSION THREAD

**Discussion by Krillle** [Kemmey/Kemmey-TeslaWatch-Public#561] [Ideas]
Hello @Kemmey. 

I have an idea how to reduce battery drain when you are away from your car (eg.g on vacation). Currently there is the travel mode, but you need to remember to enable it. 

Before requesting data from the car API, the app could check the distance between current location of the user (the watch) and the last known position of the car. If the distance is greater than let's say 100 km, it is anticipated the user is away from his car and no updates are needed. No data will be requested from the car, to let is sleep. 

This would stop battery drain as soon as you are away from your car.

**Comment by Kemmey** [comment]
That's a good suggestion - but ultimately I don't think it's necessary, as the app doesn't wake the car from the background anymore. In fact, I'm planning to completely remove the option to wake in background, as Teslas caching is working flawlessly. So the only possibility for the app to wake the car, would be opening the app. And even if you did that by accident, you would be reminded to enable travel mode. 

There's also the privacy aspect of needing the users location for this - some would be suspicious of allowing location access, not to access a feature, but solely to use the apps main screen.
